File: Lexor/UI/lexor_desktop/lib/models/dashboard_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'dashboard_response.g.dart';

@JsonSerializable()
class DashboardResponse {
  final int totalEmployees;
  final int newEmployeesThisMonth;
  final int activeContracts;
  final int expiringContractsSoon;
  final double attendanceRate;
  final double attendanceRateChange;
  final int pendingLeaves;
  final int pendingVacationLeaves;
  final List<LeavesByDayItem> leavesByDay;
  final List<LeavesByTypeItem> leavesByType;
  final List<LeavesByStatusItem> leavesByStatus;
  final HrMetricsItem hrMetrics;

  DashboardResponse({
    required this.totalEmployees,
    required this.newEmployeesThisMonth,
    required this.activeContracts,
    required this.expiringContractsSoon,
    required this.attendanceRate,
    required this.attendanceRateChange,
    required this.pendingLeaves,
    required this.pendingVacationLeaves,
    required this.leavesByDay,
    required this.leavesByType,
    required this.leavesByStatus,
    required this.hrMetrics,
  });

  factory DashboardResponse.fromJson(Map<String, dynamic> json) =>
      _$DashboardResponseFromJson(json);
}

@JsonSerializable()
class LeavesByDayItem {
  final DateTime date;
  final String dayLabel;
  final int count;

  LeavesByDayItem({
    required this.date,
    required this.dayLabel,
    required this.count,
  });

  factory LeavesByDayItem.fromJson(Map<String, dynamic> json) =>
      _$LeavesByDayItemFromJson(json);
}

@JsonSerializable()
class LeavesByTypeItem {
  final String typeName;
  final int count;

  LeavesByTypeItem({required this.typeName, required this.count});

  factory LeavesByTypeItem.fromJson(Map<String, dynamic> json) =>
      _$LeavesByTypeItemFromJson(json);
}

@JsonSerializable()
class LeavesByStatusItem {
  final String status;
  final int count;

  LeavesByStatusItem({required this.status, required this.count});

  factory LeavesByStatusItem.fromJson(Map<String, dynamic> json) =>
      _$LeavesByStatusItemFromJson(json);
}

@JsonSerializable()
class HrMetricsItem {
  final double attendanceRate;
  final double contractFillRate;
  final double leaveApprovalRate;
  final double salaryPaymentRate;
  final double activeEmployeeRate;

  HrMetricsItem({
    required this.attendanceRate,
    required this.contractFillRate,
    required this.leaveApprovalRate,
    required this.salaryPaymentRate,
    required this.activeEmployeeRate,
  });

  factory HrMetricsItem.fromJson(Map<String, dynamic> json) =>
      _$HrMetricsItemFromJson(json);
}

File: Lexor/Lexor.Model/Responses/DashboardResponse.cs
namespace Lexor.Model.Responses
{
    public class DashboardResponse
    {
        public int TotalEmployees { get; set; }
        public int NewEmployeesThisMonth { get; set; }
        public int ActiveContracts { get; set; }
        public int ExpiringContractsSoon { get; set; }
        public double AttendanceRate { get; set; }
        public double AttendanceRateChange { get; set; }
        public int PendingLeaves { get; set; }
        public int PendingVacationLeaves { get; set; }
        public List<LeavesByDayItem> LeavesByDay { get; set; } = [];
        public List<LeavesByTypeItem> LeavesByType { get; set; } = [];
        public List<LeavesByStatusItem> LeavesByStatus { get; set; } = [];
        public HrMetricsItem HrMetrics { get; set; } = new();
    }
    public class LeavesByDayItem
    {
        public DateTime Date { get; set; }
        public string DayLabel { get; set; } = string.Empty;
        public int Count { get; set; }
    }
    public class LeavesByTypeItem
    {
        public string TypeName { get; set; } = string.Empty;
        public int Count { get; set; }
    }
    public class LeavesByStatusItem
    {
        public string Status { get; set; } = string.Empty;
        public int Count { get; set; }
    }

    public class HrMetricsItem
    {
        public double AttendanceRate { get; set; }
        public double ContractFillRate { get; set; }
        public double LeaveApprovalRate { get; set; }
        public double SalaryPaymentRate { get; set; }
        public double ActiveEmployeeRate { get; set; }
    }
}
